packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -734,32 +736,113 @@ export class Stack extends Construct implements ITaggable {
    *
    * @internal
    */
-  public _addAssemblyDependency(target: Stack, reason?: string) {
+  public _addAssemblyDependency(target: Stack, reason: StackDependencyReason = { description: '' }) {
```

Comment:
Why is description blank?

Reply:
Since `reason` used to be optional and `reason.description` is the closest thing to a replacement, I moved to providing a blank default value. I could default it to something like `${source.node.path} depends on ${target.node.path}`, though that doesn't really add any information, since the `source` and `target` are already included in `reason`, now.

Reply:
I think this needs to be modeled in a way that a blank string isn't the default input. Description should likely be optional where this is defined.